pkg/plugincmdtree/plugins_cache.go

```diff
@@ -0,0 +1,257 @@
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal plugin command tree")
+	}
+	err = os.WriteFile(GetPluginsCommandTreeCachePath(), data, 0644)
+	if err != nil {
+		return errors.Wrapf(err, "failed to write the command tree to %q file ", GetPluginsCommandTreeCachePath())
+	}
+	return nil
+}
+
+func (c *cacheImpl) constructPluginCommandTree(plugin *cli.PluginInfo) (*CommandNode, error) {
+	if err := c.pluginDocsGenerator(plugin); err != nil {
+		return nil, errors.Wrapf(err, "failed to generate docs for the plugin %q", plugin.Name)
+	}
+	// construct the command tree
+	cmdTreeRoot := NewCommandNode()
+
+	docsDir := getPluginsDocsCachePath()
+	files, err := os.ReadDir(docsDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error while reading local plugin command tree directory")
+	}
+	var aliasErrGroup errgroup.Group
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		// ignore non mark down files
+		if filepath.Ext(file.Name()) != ".md" {
+			continue
+		}
+
+		filename := strings.TrimSuffix(file.Name(), ".md")
+		cmdNames := strings.Split(filename, "_")
+
+		var aliasArgs []string
+		current := cmdTreeRoot
+		for _, cmdName := range cmdNames {
+			if current.Subcommands[cmdName] == nil {
+				current.Subcommands[cmdName] = NewCommandNode()
+			}
+
+			current = current.Subcommands[cmdName]
+			if cmdName != "tanzu" && cmdName != plugin.Name {
+				aliasArgs = append(aliasArgs, cmdName)
+				aliasArgsCopy := make([]string, len(aliasArgs))
+				copy(aliasArgsCopy, aliasArgs)
+				if !current.AliasProcessed {
+					// kickoff the goroutine to add the alias to the command
+					aliasErrGroup.Go(func() error {
+						cmdAlias, aliasErr := getPluginCommandAlias(plugin, aliasArgsCopy)
+						if aliasErr != nil {
+							return aliasErr
+						}
+						current.Aliases = cmdAlias
+						return nil
+					})
+					current.AliasProcessed = true
+				}
+			}
+		}
+	}
+	// Wait for all goroutines to finish or one of them to return an error
+	if err := aliasErrGroup.Wait(); err != nil {
+		return nil, errors.Wrap(err, "failed to generate command alias")
+	}
+	if cmdTreeRoot.Subcommands["tanzu"] != nil && cmdTreeRoot.Subcommands["tanzu"].Subcommands[plugin.Name] != nil {
+		return cmdTreeRoot.Subcommands["tanzu"].Subcommands[plugin.Name], nil
+	}
+
+	return nil, nil
+}
+
+func getPluginCommandTree() (*pluginCommandTree, error) {
+	b, err := os.ReadFile(GetPluginsCommandTreeCachePath())
+	if err != nil {
+		return &pluginCommandTree{
+			CommandTree: make(map[pluginKey]*CommandNode),
+		}, nil
+	}
+
+	var ctr pluginCommandTree
+	err = yaml.Unmarshal(b, &ctr)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal the  plugin command tree ")
+	}
+
+	return &ctr, nil
+}
+
+func generatePluginDocs(plugin *cli.PluginInfo) error {
+	docsDir := getPluginsDocsCachePath()
+	_ = os.RemoveAll(docsDir)
+	_ = os.MkdirAll(docsDir, 0755)
+
+	args := []string{"generate-docs", "--docs-dir", docsDir}
```

Comment:
It's hard to explain to the curious user what the doc/ subdir is, since contents come and go based on what the latest plugins we invoked the generate-docs command on. How about hiding it using a name like .docs?

Reply:
Good point! Updated the docs path to use `.docs`